src/components/TreeView/TreeView.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+// import { classNames } from '@shopify/react-utilities/styles';
+
+import { SourceData } from './interface';
+import TreeNode from './TreeNode';
+import { TREEVIEW } from '../ThemeIdentifiers';
+
+import * as baseTheme from './TreeView.scss';
+
+// There could be multiple themes, but right now lets take only a basic theme
+type Themes = 'basic';
+
+export interface Props {
+  selectedTheme?: Themes;
+  source: SourceData[];
+  theme?: any;
+}
+
+export interface State {
+  source: SourceData[];
+}
+
+class TreeView extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      source: this.props.source,
+    };
+  }
+
+  componentWillReceiveProps(newProps: Props) {
+    // If treenodes are changed or deleted or added new then it should render again
+    if (!Object.is(newProps.source, this.props.source)) {
+      this.setState({ source: newProps.source });
+    }
+  }
+
+  // Set the active or inactive status of node
+  toggleNode = (id: number) => {
+    const { source } = this.props;
+
+    source.some((item: SourceData): any => {
+      if (item.id === id) {
+        item.active = !item.active;
+        // Call the toggle callback if available
+        if (item.onToggle) {
+          item.onToggle(item.active);
+        }
+
+        this.setState({ source });
+        return true;
+      }
+
+      // If current node have children then search this id in child node
+      if (item.children) {
+        if (this.toggleChildNode(item.children, id)) {
+          this.setState({ source });
+          return true;
+        }
+      }
+    });
+  }
+
+  // Set child nodes active or inactive status
+  toggleChildNode = (source: SourceData[], id: number) => {
+    return source.some((item: SourceData): any => {
```

Comment:
I believe you don't have to specify `: any` here.  If you do, i think it should be a boolean.  Let's not use `any` unless we absolutely have to.